src/functions/listFunctions.ts

```diff
@@ -0,0 +1,60 @@
+import Table = require("cli-table");
+import * as backend from "../deploy/functions/backend";
+import { logger } from "../logger";
+import { previews } from "../previews";
+import * as args from "../deploy/functions/args";
+
+/**
+ * Lists all functions in the Firebase project
+ * @param context the Context of the project
+ * @returns mapping that contains a list of functions under the 'functions' key
+ */
+export async function listFunctions(
+  context: args.Context
+): Promise<{ functions: backend.FunctionSpec[] }> {
+  const bkend = await backend.existingBackend(context, true);
+  const functionSpecs = previews.functionsv2
+    ? bkend.cloudFunctions
+    : bkend.cloudFunctions.filter((fn) => fn.platform === "gcfv1");
+  functionSpecs.sort((fn1, fn2) => {
+    if (fn1.entryPoint > fn2.entryPoint) {
+      return 1;
+    }
+    if (fn1.entryPoint < fn2.entryPoint) {
+      return -1;
+    }
+    return 0;
+  });
+  const table = previews.functionsv2
+    ? new Table({
+        head: ["Function", "Version", "Trigger", "Location", "Memory", "Runtime"],
+        style: { head: ["yellow"] },
+      })
+    : new Table({
+        head: ["Function", "Trigger", "Location", "Memory", "Runtime"],
+        style: { head: ["yellow"] },
+      });
+  for (const fnSpec of functionSpecs) {
+    const trigger = backend.isEventTrigger(fnSpec.trigger) ? fnSpec.trigger.eventType : "https";
```

Comment:
I wonder if we want something more clever than just this. E.g. should we call a scheduled function "scheduled" instead of "google.pubsub.topic.publish" and should we call a callable function "callable" instead of "https".

WDYT @taeold?

Reply:
that sounds like a good ux improvement, but I'd be happy with shipping what we have here and tackling this as a follow up work to pick up later if its going to take more than few hours.

Reply:
We can create a mapping for trigger types from full name to shorthand, like "google.pubsub.topic.publish" -> "scheduled". Is there an internal doc somewhere that has similar shorthand mappings like this?

Also, since this is a user facing change, would this need to go back through council?

Reply:
Sounds good, I'll make a feature request for creating this mapping and assign myself